pkg/catalogv2/git/git.go

```diff
@@ -1,480 +1,395 @@
+
+		return nil
 	}
 
-	return client, nil
+	// if all else failed, something nasty happened
+	return errors.New("could not set repository credentials")
 }
 
-func (g *git) remoteSHAChanged(branch, sha string) (bool, error) {
-	formattedURL := formatGitURL(g.URL, branch)
-	if formattedURL == "" {
-		return true, nil
+// setRepoOptions assigns the options configured before in credentials.
+// hard-code other needed configurations like Depth for faster cloning.
+func (r *Repository) setRepoOptions() {
+	// Clone Options
+	r.cloneOpts.URL = r.URL
+	r.cloneOpts.Depth = 1
+	r.cloneOpts.InsecureSkipTLS = r.insecureTLSVerify
+	r.cloneOpts.Tags = gogit.NoTags
+
+	// Fetch Options
+	r.fetchOpts.RemoteURL = r.URL
+	r.fetchOpts.Depth = 1
+	r.fetchOpts.Force = true
+	r.fetchOpts.InsecureSkipTLS = r.insecureTLSVerify
+	r.fetchOpts.Tags = gogit.NoTags
+
+	// List Options
+	r.listOpts.InsecureSkipTLS = r.insecureTLSVerify
+
+	// Reset Options
+	r.resetOpts.Mode = gogit.HardReset
+
+	// Set authentication Methods for cloning
+	if r.auth != nil {
+		r.cloneOpts.Auth = r.auth
+		r.fetchOpts.Auth = r.auth
+		r.listOpts.Auth = r.auth
+	}
+	// Set CABundle for all git operations
+	if len(r.caBundle) > 0 {
+		r.cloneOpts.CABundle = r.caBundle
+		r.fetchOpts.CABundle = r.caBundle
+		r.listOpts.CABundle = r.caBundle
+	}
+	// Debug if enabled
+	if logrus.IsLevelEnabled(logrus.DebugLevel) {
+		r.cloneOpts.Progress = os.Stdout
+		r.fetchOpts.Progress = os.Stdout
 	}
+}
 
-	client, err := g.httpClientWithCreds()
-	if err != nil {
-		logrus.Warnf("Problem creating http client to check git remote sha of repo [%v]: %v", g.URL, err)
-		return true, nil
+// cloneOrOpen executes the clone operation of a git repository at given branch with depth = 1 if it does not exist.
+// If exists, just open the local repository and assign it to ro.localRepo
+func (r *Repository) cloneOrOpen(branch string) error {
+	var err error
+	cloneOptions := r.cloneOpts
+	if branch != "" {
+		cloneOptions.ReferenceName = plumbing.NewBranchReferenceName(branch)
 	}
-	defer client.CloseIdleConnections()
 
-	req, err := http.NewRequest("GET", formattedURL, nil)
+	err = cloneOptions.Validate()
 	if err != nil {
-		logrus.Warnf("Problem creating request to check git remote sha of repo [%v]: %v", g.URL, err)
-		return true, nil
+		return fmt.Errorf("plainClone validation failure: %w", err)
 	}
 
-	req.Header.Set("Accept", "application/vnd.github.v3.sha")
-	req.Header.Set("If-None-Match", fmt.Sprintf("\"%s\"", sha))
-	for k, v := range g.headers {
-		req.Header.Set(k, v)
-	}
-
-	resp, err := client.Do(req)
-	if err != nil {
-		// Return timeout errors so caller can decide whether or not to proceed with updating the repo
-		uErr := &url.Error{}
-		if ok := errors.As(err, &uErr); ok && uErr.Timeout() {
-			return false, errors.Wrapf(uErr, "Repo [%v] is not accessible", g.URL)
+	err = r.plainOpen()
+	if err == gogit.ErrRepositoryNotExists {
+		r.repoGogit, err = gogit.PlainClone(r.Directory, false, cloneOptions)
```

Comment:
Instead of `git.PlainClone`, I would use `git.Clone` instead and leverage [BoundOS](https://github.com/go-git/go-billy/pull/31/commits/3c59de8dc969175c9164e56fe768111cc5c74a0e) from the latest `go-billy` version.

The main reason why I would do that is that, in specific scenarios, the default `osfs` implementation will cause the worktree to always appear to be "dirty" even when no files have changed. This would happen when a) the repository contains absolute symlinks pointing to outside of the "chroot" or b) it contains files with non regular file modes.

It would look something like this:

```suggestion
		wt := osfs.New(r.Directory, osfs.WithBoundOS())
		storer := filesystem.NewStorage(osfs.New(filepath.Join(r.Directory, gogit.GitDirName), osfs.WithBoundOS()), cache.NewObjectLRUDefault())
		r.repoGogit, err = gogit.Clone(storer, wt, cloneOptions)
```